include/openthread/platform/dnssd.h

```diff
@@ -0,0 +1,425 @@
+    const uint8_t     *mTxtData;             ///< Encoded TXT data bytes.
+    uint16_t           mTxtDataLength;       ///< Length of TXT data.
+    uint16_t           mPort;                ///< The service port number.
+    uint16_t           mPriority;            ///< The service priority.
+    uint16_t           mWeight;              ///< The service weight.
+    uint32_t           mTtl;                 ///< The service TTL in seconds.
+    uint32_t           mInfraIfIndex;        ///< The infrastructure network interface index.
+} otPlatDnssdService;
+
+/**
+ * Represent a DNS-SD host.
+ *
+ * See `otPlatDnssdRegisterHost()`, `otPlatDnssdUnregisterHost()` for more details about fields in each case.
+ *
+ */
+typedef struct otPlatDnssdHost
+{
+    const char         *mHostName;        ///< The host name (does not include domain name).
+    const otIp6Address *mAddresses;       ///< Array of IPv6 host addresses.
+    uint16_t            mAddressesLength; ///< Number of entries in @p mAddresses array.
+    uint32_t            mTtl;             ///< The host TTL in seconds.
+    uint32_t            mInfraIfIndex;    ///< The infrastructure network interface index.
+} otPlatDnssdHost;
+
+/**
+ * Represent a DNS-SD key record.
+ *
+ * See `otPlatDnssdRegisterKey()`, `otPlatDnssdUnregisterKey()` for more details about fields in each case.
+ *
+ */
+typedef struct otPlatDnssdKey
+{
+    const char    *mName;          ///< A host or a service instance name (does not include domain name).
+    const char    *mServiceType;   ///< The service type if key is for a service (does not include domain name).
+    const uint8_t *mKeyData;       ///< Byte array containing the key record data.
+    uint16_t       mKeyDataLength; ///< Length of @p mKeyData in bytes.
+    uint16_t       mClass;         ///< The resource record class.
+    uint32_t       mTtl;           ///< The TTL in seconds.
+    uint32_t       mInfraIfIndex;  ///< The infrastructure network interface index.
+} otPlatDnssdKey;
+
+/**
+ * Callback from platform to notify OpenThread of state changes to DNS-SD module.
+ *
+ * OpenThread stack will call `otPlatDnssdGetState()` (from this callback or later) to get the new state. The platform
+ * MUST therefore ensure that the returned state from `otPlatDnssdGetState()` is updated before calling this.
+ *
+ * @param[in] aInstance The OpenThread instance structure.
+ *
+ */
+extern void otPlatDnssdStateHandleStateChange(otInstance *aInstance);
+
+/**
+ * Gets the current state of DNS-SD module.
+ *
+ * Platform MUST notify OpenThread whenever its state gets changed by invoking `otPlatDnssdStateHandleStateChange()`.
+ *
+ * @param[in] aInstance The OpenThread instance.
+ *
+ * @returns The current state of DNS-SD module.
+ *
+ */
+otPlatDnssdState otPlatDnssdGetState(otInstance *aInstance);
+
+/**
+ * Registers or updates a service on the infrastructure network's DNS-SD module.
```

Comment:
All the suggestions in this comments for this method, can you also apply the same things to the rest of the methods here?  Mainly

OpenThread stack --> The OpenThread stack

and indicating "field" in each field bullet

Reply:
Thanks. 👍. Made all the changes.